src/plugins/intel_npu/src/compiler/src/graph_transformations.cpp

```diff
@@ -6,29 +6,53 @@
 
 #include <istream>
 #include <mutex>
+#include <streambuf>
 
 #include "openvino/pass/serialize.hpp"
 #include "transformations/op_conversions/convert_interpolate11_downgrade.hpp"
 
 namespace intel_npu::driverCompilerAdapter {
 
-IR serializeToIR(const std::shared_ptr<const ov::Model>& origModel, uint32_t supportedOpset) {
+IR::IR(const std::shared_ptr<const ov::Model>& origModel, uint32_t supportedOpset, bool largeModel)
+    : _logger("LevelZeroCompilerAdapter::IR", Logger::global().level()),
+      _isLargeModel(largeModel),
+      _xmlStream(&_xmlCache),
+      _weightsStream(&_weightsCache) {
     // There is no const variant of run_passes so use const_cast here
     // as model serialization does not mutate the model
-    std::shared_ptr<ov::Model> model = std::const_pointer_cast<ov::Model>(origModel);
+    _model = std::const_pointer_cast<ov::Model>(origModel);
+
+#ifdef _WIN32
+    // Only use fstream for Windows
```

Comment:
"custom streambuf" instead of "fstream"?
Should we add a MAX_WIN_STREAMBUF_SIZE define?

Reply:
Use INT32_MAX as value of MAX_WIN_STREAMBUF_SIZE now. Thanks.